pages/api/retrieval.ts

```diff
@@ -0,0 +1,117 @@
+  let submissionIDlist: (string | undefined)[] = [];
+  let formResponses: QueryCommandOutput;
+  let db: DynamoDBClient;
+  try {
+    //Create dynamodb client
+    db = new DynamoDBClient({ region: process.env.AWS_REGION ?? "ca-central-1" });
+    //Create form's responses db param
+    //NB: A filter expression is applied after a Query finishes, but before the results are returned.
+    //Therefore, a Query consumes the same amount of read capacity, regardless of whether a filter expression is present.
+    const getItemsDbParams = {
+      TableName: "Vault",
+      IndexName: "retrieved-index",
+      Limit: expectedMaxRecords,
+      //Cannot use partitin key or sort key such as formID and Retrieved attribute in keyConditionExpression simultaneously
+      KeyConditionExpression: "Retrieved = :zero",
+      ExpressionAttributeValues: { ":formID": { S: formID }, ":zero": { N: "0" } },
+      //A filter expression cannot contain partition key or sort key attributes.
+      //You need to specify those attributes in the key condition expression, not the filter expression.
+      FilterExpression: "FormID = :formID",
+      ProjectionExpression: "FormID,SubmissionID,FormSubmission,Retrieved",
+    };
+    //Get form's responses for formID
+    formResponses = await db.send(new QueryCommand(getItemsDbParams));
+  } catch (error) {
+    logMessage.error(error);
+    return res
+      .status(500)
+      .json({ responses: "Error on Server Side when fetching form's responses" });
+  }
+  try {
+    //Collecting items submissionIDs for logging and updating items.
+    submissionIDlist =
+      formResponses?.Items?.map((response) => {
+        return response.SubmissionID.S;
+      }) ?? [];
+    if (formResponses.Count && formResponses.Count > 0) {
+      //Preparing a query to set records's attribute Retrieved to 1.
+      for (const submissionID of submissionIDlist as string[]) {
+        //Create a new update object statement
+        const updateItem = {
+          TableName: "Vault",
+          Key: {
+            SubmissionID: {
+              S: submissionID,
+            },
+            FormID: {
+              S: formID,
+            },
+          },
+          UpdateExpression: "SET Retrieved = :retrieved",
+          ConditionExpression: "SubmissionID = :submission",
+          ExpressionAttributeValues: {
+            ":retrieved": { N: "1" },
+            ":submission": { S: submissionID },
+          },
+          ReturnValues: "NONE",
+        };
+        //Update one item at the time
+        await db?.send(new UpdateItemCommand(updateItem));
+      }
+    }
+    //Return responses data
+    return res.status(200).json({ responses: formResponses.Items });
+  } catch (error) {
+    logMessage.warn(`Possible data loss for these submissionIDs`);
+    //print submission ids as string
+    logMessage.warn(submissionIDlist ? submissionIDlist.toString() : "Empty submissionID list");
+    logMessage.error(error);
```

Comment:
The above logs could be concatenated into one string before logged. With a busy system, it's possible that other `logMessage` commands could be executed by another client, and these messages could be split up.